cli/src/main/java/com/devonfw/tools/ide/process/ProcessContextImpl.java

```diff
@@ -272,4 +248,90 @@ private String findBashOnWindows() {
+
+    try (BufferedReader outReader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+        BufferedReader errReader = new BufferedReader(new InputStreamReader(process.getErrorStream()))) {
+      String outLine = "";
+      String errLine = "";
+      while ((outLine != null) || (errLine != null)) {
+        if (outLine != null) {
+          outLine = outReader.readLine();
+          if (outLine != null) {
+            out.add(outLine);
+          }
+        }
+        if (errLine != null) {
+          errLine = errReader.readLine();
+          if (errLine != null) {
+            err.add(errLine);
+          }
+        }
+      }
+    }
+  }
+
+  private boolean checkAndHandlePossibleBashScript(String executableName) {
+
+    String fileExtension = FilenameUtil.getExtension(executableName);
+    boolean isBashScript = "sh".equals(fileExtension) || hasSheBang(this.executable);
+    if (isBashScript) {
+      String bash = "bash";
+      if (this.context.getSystemInfo().isWindows()) {
+        String findBashOnWindowsResult = findBashOnWindows();
+        if (findBashOnWindowsResult != null) {
+          bash = findBashOnWindowsResult;
+        }
+      }
+      this.arguments.add(0, bash);
+    }
+    return isBashScript;
+  }
+
+  private void performLogOnError(ProcessResult result, int exitCode) {
+
+    if (!result.isSuccessful() && (this.errorHandling != ProcessErrorHandling.NONE)) {
+      String message = createCommandMessage(" failed with exit code " + exitCode + "!");
+      if (this.errorHandling == ProcessErrorHandling.THROW) {
+        throw new CliException(message, exitCode);
+      }
+      IdeSubLogger level;
+      if (this.errorHandling == ProcessErrorHandling.ERROR) {
+        level = this.context.error();
+      } else if (this.errorHandling == ProcessErrorHandling.WARNING) {
+        level = this.context.warning();
+      } else {
+        level = this.context.error();
+        level.log("Internal error: Undefined error handling {}", this.errorHandling);
+      }
+      level.log(message);
+    }
+  }
+
+  private void modifyArgumentsOnBackgroundProcess(boolean isBashScript) {
+
+    if (context.getSystemInfo().isWindows() && !isBashScript) {
+
+      this.context.warning(
+          "Currently starting the process as background process in windows will use 'more' command to redirect output to a new cmd window!");
+
+      this.processBuilder.redirectOutput(Redirect.PIPE).redirectError(Redirect.PIPE);
+      List<String> windowsArgs = List.of("cmd.exe", "/c", "start", "cmd.exe", "/k");
+
+      String currentCommandToRunInCmd = this.arguments.stream().map(Object::toString).collect(Collectors.joining(" "));
+      currentCommandToRunInCmd += " | more";
+
+      this.arguments.clear();
+
+      List<String> newArgs = new ArrayList<>(windowsArgs);
+      newArgs.add(currentCommandToRunInCmd);
+
+      this.arguments.addAll(newArgs);
+
```

Comment:
Maybe we can do it like this and avoid 2 new local variables? let me know what you think: 
```
      addArgs("|", "more");
      List<String> windowsArgs = List.of("cmd.exe", "/c", "start", "cmd.exe", "/k");
      for (int i = windowsArgs.size() - 1; i >= 0; i--) {
        this.arguments.add(0, windowsArgs.get(i));
      }
```

Reply:
I would rather prefer simple to understand code than trying reducing variables. I think a reversed loop would make it harder to understand what is done.

Reply:
You're right, the loop makes it ugly, maybe like this?: 
```
      addArgs("|", "more");
      List<String> windowsArgs = List.of("cmd.exe", "/c", "start", "cmd.exe", "/k");
      List<String> newArgs = new ArrayList<>(windowArgs);
      newArgs.addAll(this.args);
      this.args = newArgs;
```

Reply:
Concatenating the passed arguments is currently required, since not the subprocess starts the application, rather the cmd.exe will start our background process independently.